### DISCUSSION TITLE
Running Duino PC_Miner.py permanently

### DISCUSSION THREAD

**Discussion by mh60385** [revoxhere/duino-coin#689] [Q&A]
Hi, 
I run a RPI 4 headless. 
Would like to run the Duino Miner continuously but when i ssh out of my rpi the window closes and kills the process. 
Any suggestions for getting this to run continuously

Chers MB

**Comment by naphob** [comment]
you can use screen or nohup over ssh

**Reply by JamesHFredPH** [reply]
you want to make sure 'screen' is loaded...

screen is, 'I think', that original interactive CLI Session Manager.
(Or one of the Other Variants, that you know how to use...)

sudo apt install screen

Then you can SSH into your Pi, or create a Startup Script.

In a Comman Line all at once, or in a Shell Script.
     /usr/bin/screen -d -m  /usr/bin/python3 'PC_Miner.py'

The above will start the PC_Miner.py in a detached CLI session.

For Me it seemed to work better to explicitly define App/Script Locations.





Interactively...
     Get to the PC_Miner.py Directory.
 
     Type:
          screen
     (this will start a NEW screen session)

    Type:
          python3 PC_Miner.py
     then press CTRL-a then d to detach the session
     (make sure 'Caplocks' is not on.)

    Type:
         screen -list

    to see your detached session(s) (it should say detached)

    Type:
          screen -r xxxxx 
    (where xxxxx is the session number will reattach)
    
    Then CTRL-a then d 
    (Will detach again)

    The one thing that I see as a PAIN in screen is that 'it's default action is to create a new screen session'.
    So if you are NOT paying attention your can 'stack up screen sessions and it can become an issue.

When you 'Detach your CLI Session' and then Logout of the SSH your 'Detached CLI Session' will continue to run and you can 'reattach' if you log back in through SSH.  (or Locally, or however...)

**Comment by Traction6131** [comment]
As already said, you can install screen with

`sudo apt install screen`
`screen -S miner `
(for example, just the name for the session) 
And then happy mining!

**Comment by haxorcitoz** [comment]
And dont forget add this line of code at the bottom of your crontab so when the RPI reboot it automatically start mining.
@reboot screen -dm bash -c "cd ~/duino-coin; python3 PC_Miner.py"
the line "cd ~/duino-coin; python3 PC_Miner.py" depends on the path you put the PC_Miner.py. For me its on ~/duino-coin
